auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,126 @@
+            self.env["auditlog.rule"]
+            .sudo()
+            .search([("model_id.model", "=", model_name)], limit=1)
+        )
+        if rule.auditlog_line_access_rule_ids:
+            return rule.mapped("auditlog_line_access_rule_ids.field_ids.name")
+        return []
+
+    @api.model
+    @tools.ormcache("model_name")
+    def _get_log_selected_fields_only(self, model_name):
+        """Memory-cached translation of model to rule"""
+        rule = (
+            self.env["auditlog.rule"]
+            .sudo()
+            .search([("model_id.model", "=", model_name)], limit=1)
+        )
+        return rule.log_selected_fields_only
+
+    @api.model
+    def get_auditlog_fields(self, model):
+        res = super(AuditlogRule, self).get_auditlog_fields(model)
+        if self._get_log_selected_fields_only(model._name):
+            selected_field_names = self._get_field_names_of_rule(model._name)
+            # we re-use the checks on non-stored fields from super.
+            res = [x for x in selected_field_names if x in res]
+        return res
+
+    def write(self, values):
+        cache_invalidating_fields = [
+            "state",
+            "auditlog_line_access_rule_ids",
+            "log_selected_fields_only",
+        ]
+        if any([field in values.keys() for field in cache_invalidating_fields]):
+            # clear cache for all ormcache methods.
+            self.clear_caches()
+        return super(AuditlogRule, self).write(values)
+
+    @api.onchange("model_id")
+    def onchange_model_id(self):
+        # if model changes we must wipe out all field ids
+        self.auditlog_line_access_rule_ids.unlink()
+
+    @api.model
+    def _get_view_log_lines_action(self):
+        assert self.env.context.get("active_model")
+        assert self.env.context.get("active_ids")
+        model = (
+            self.env["ir.model"]
+            .sudo()
+            .search([("model", "=", self.env.context.get("active_model"))])
+        )
+        domain = [
+            ("model_id", "=", model.id),
+            ("res_id", "in", self.env.context.get("active_ids")),
+        ]
+        return {
+            "name": _("View Log Lines"),
+            "res_model": "auditlog.log.line",
+            "view_mode": "tree,form",
+            "view_id": False,
+            "domain": domain,
+            "type": "ir.actions.act_window",
+        }
+
+    def _create_server_action(self):
+        self.ensure_one()
+        code = "action = env['auditlog.rule']._get_view_log_lines_action()"
+        allowed_groups = self.auditlog_line_access_rule_ids.group_ids
+        server_action = (
+            self.env["ir.actions.server"]
+            .sudo()
+            .create(
+                {
+                    "name": "View Log Lines",
+                    "model_id": self.model_id.id,
+                    "state": "code",
+                    "code": code,
+                    "groups_id": [(6, 0, allowed_groups.ids)],
```

Comment:
```suggestion
                    "groups_id": [(6, 0, allowed_groups.ids)] if not any(not x.group_ids for x in self.auditlog_line_access_rule_ids),
```

if there's an unrestricted access rule and a restricted one, all users should see the action. ir.rules will take care of what they see then

Reply:
In that sense then we do not need to assign any groups_id here. We just need to make sure that the user is also an auditlog user

Reply:
... which is admin work